internal/testing/mockClearlyDefined/mock_clearly_defined.go

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockclearlydefined
```

Comment:
could you also change the tests in the scanner (https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/common/scanner/scanner_test.go#L343)?

Reply:
Sure!

Reply:
I've added in the mock server and tests are passing for me locally. I think it is a good pattern, but not sure if I am missing anything. Feel free to suggest alternative ways to get a testing client into the scanner!